pinot-common/pom.xml

```diff
@@ -372,5 +337,45 @@
         <shade.phase.prop>package</shade.phase.prop>
       </properties>
     </profile>
+    <profile>
+      <!-- These plugins don't care about unchanged (re)sources and will always generate.
+      This causes the maven-compiler-plugin to detect changes and always recompile the Java sources
+      If there are changes in src/main/codegen, either run mvn clean or mvn -P sqlparser
+       -->
+      <id>sqlparser</id>
+      <activation>
+        <file>
+          <missing>target/generated-sources/fmpp/javacc/Parser.jj</missing>
+        </file>
```

Comment:
Cool! I didn't think about using this to enable the profile after a clean start.

Reply:
I thought of other solutions as well, but that always meant 2 separate steps to execute, and probably hard to understand.
Be aware that this is also a bit tricky: `mvn clean verify` won't work, because profile activation is decided before plugin execution. If you do this, the profile will not be activated (because the file still exists), clean will remove it and compiler-plugin will fail because of a missing class.
If you run it again it works fine again :)
Anyway, this is most likely the simplest hack that will do its job good enough.

Reply:
Can you add a comment in this activation to inform about the `mvn clean verify` limitation?

Reply:
done